### PR TITLE
Use the new V2 API

### PR DESCRIPTION
**Pull request by dabroek**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -77,7 +77,11 @@
     "transform": {
       "^.+\\.jsx?$": "babel-jest",
       "^.+\\.pem$": "<rootDir>/tests/unit/__stubs__/fileTransformer.js"
-    }
+    },
+    "testPathIgnorePatterns": [
+      "/node_modules/",
+      "tests/integration/refunds.test.js"
```
Comment (dabroek) [inline package.json]: Temporary, until the refunds endpoint is activated.

File: tests/integration/payments.test.js [hunk 2]
```diff
@@ -39,12 +39,14 @@ describe('payments', () => {
         }
 
         paymentExists.then((payment) => {
-          /**
-           * If you want to test the rest of this flow you have to
-           * manually set this payment to paid.
-           */
           if (!payment.isPaid()) {
-            console.log('If you want to test the full flow, set the payment to paid:', payment.links.paymentUrl);
+            console.log('If you want to test the full flow, set the payment to paid:', payment.getPaymentUrl());
```
Comment (vernondegoede) [inline tests/integration/payments.test.js]: This is cool! 🤘

File: src/models/payment.js [hunk 3]
```diff
@@ -22,30 +22,50 @@ export default class Payment extends MollieModel {
...
+      canceledAt: null,
+      expiresAt: null,
+      expiredAt: null,
+      failedAt: null,
+      amount: {
+        value: null,
+        currency: null,
+      },
       amountRefunded: null,
       amountRemaining: null,
```
[the remark this answers is not in the document]
Comment (dabroek) [reply src/models/payment.js]: The `applicationFee`, `amountRefunded`, `amountRemaining`, and `settlementAmount` are not guaranteed to be included in the response. Therefore these fields should stay `null` and not represent an `amount object`.

[the remark this answers is not in the document]
Comment (vernondegoede) [reply src/models/payment.js]: Ok. Makes sense 👍

**Comment by lvgunst** [comment]
The example in the readme still uses `amount: 10.00` and this needs to be a object for multi currency.

**Comment by lvgunst** [comment]
I can't get it to work:

```js
mollie.payments.create({
    amount: {
        value: '10.00',
        currency: 'EUR',
    },
    description: 'My first API payment',
    redirectUrl: 'https://yourwebshop.example.org/order/123456',
    webhookUrl: 'https://yourwebshop.example.org/webhook'
}, function (err, data) {
    console.log('err', err); // err is undefined
    console.log('data', data); // data is undefined
});
```

```js
mollie.payments.create({
    amount: {
        value: '10.00',
        currency: 'EUR',
    },
    description: 'My first API payment',
    redirectUrl: 'https://yourwebshop.example.org/order/123456',
    webhookUrl: 'https://yourwebshop.example.org/webhook'
})
    .then((payment) => {
        // Forward the customer to the payment.getPaymentUrl()
        console.log('payment', payment);
    })
    .catch((err) => {
        console.log('error', err); // err is undefined
    });
```

**Comment by vernondegoede** [comment]
@lvgunst What does you `mollie` variable look like? You need to instantiate a new client. This works for me:

```
const mollieClient = require("./dist/mollie");
const mollie = mollieClient({ apiKey: "<my_api_key>" });

mollie.payments
  .create({
    amount: {
      value: "10.00",
      currency: "EUR",
    },
    description: "My first API payment",
    redirectUrl: "https://yourwebshop.example.org/order/123456",
    webhookUrl: "https://yourwebshop.example.org/webhook",
  })
  .then(payment => {
    // Forward the customer to the payment.getPaymentUrl()
    console.log("payment", payment);
  })
  .catch(err => {
    console.log("error", err); // err is undefined
  });
```

**Comment by lvgunst** [comment]
I had changed the API endpoint to https://api.mollie.dev/v2. The error contained `code: 'SELF_SIGNED_CERT_IN_CHAIN',` but the errorHandler in resource.js only works with Mollie error responses.

To ignore self signed certificates I had to add an option `rejectUnauthorized: false` to `options.httpsAgent = new https.Agent()` but it currently not configurable.

**Comment by vernondegoede** [comment]
@lvgunst There's already a PR that contains some examples. We need to update these to work with the new V2 client, though.

https://github.com/mollie/mollie-api-node/pull/48

**Comment by vernondegoede** [comment]
Examples are updated. We should merge them into this branch and then merge this one into master.

**Comment by vernondegoede** [comment]
I added pagination to this branch. Updated the `README` and `MIGRATION` guide to explain how it works. Also added the examples to this branch.

**Pull request merged**